lessons/specifics/iex_helpers.md

```diff
@@ -0,0 +1,228 @@
+pop/2                pop/3                pop_lazy/3
+put/3                put_new/3            put_new_lazy/3
+split/2              take/2               to_list/1
+update!/3            update/4             values/1
+
+iex> h Map.merge/2
+                             def merge(map1, map2)
+
+Merges two maps into one.
+
+All keys in map2 will be added to map1, overriding any existing one.
+
+If you have a struct and you would like to merge a set of keys into the struct,
+do not use this function, as it would merge all keys on the right side into the
+struct, even if the key is not part of the struct. Instead, use
+Kernel.struct/2.
+
+Examples
+
+┃ iex> Map.merge(%{a: 1, b: 2}, %{a: 3, d: 4})
+┃ %{a: 3, b: 2, d: 4}
+```
+
+And as you can see, we not only found out what was offered by the module but exactly how to work what we were after with a working example!
+
+### `i`
+Since we already have another tool to learn about this and its valid Elixir code, why not use `h` to learn about `i`? Here we go:
+
+```elixir
+iex> h i
+
+                                  def i(term)
+
+Prints information about the given data type.
+
+iex> i Map
+Term
+  Map
+Data type
+  Atom
+Module bytecode
+  /usr/local/Cellar/elixir/1.3.3/bin/../lib/elixir/ebin/Elixir.Map.beam
+Source
+  /private/tmp/elixir-20160918-33925-1ki46ng/elixir-1.3.3/lib/elixir/lib/map.ex
+Version
+  [9651177287794427227743899018880159024]
+Compile time
+  no value found
+Compile options
+  [:debug_info]
+Description
+  Use h(Map) to access its documentation.
+  Call Map.module_info() to access metadata.
+Raw representation
+  :"Elixir.Map"
+Reference modules
+  Module, Atom
+```
+So now we have a bunch more information about `Map`, including all the way down to where its source is stored and the modules it references!
+This is quite useful when exploring custom and foreign data types and new functions.
+
+### `r`
+`r` is our helper for recompiling a given module.
+Say you have changed some code and simply want to run the new function you added, you could save the file then run:
+
+```elixir
+iex> r MyProject
+warning: redefining module MyProject (current version loaded from _build/dev/lib/my_project/ebin/Elixir.MyProject.beam)
+  lib/my_project.ex:1
+
+{:reloaded, MyProject, [MyProject]}
+```
+
+### `s`
+`s` gives us type spec information for a given module or function.
+This way we can know what it is expecting:
+
+```elixir
+iex> s Map.merge/2
+@spec merge(map(), map()) :: map()
+
+# it also works on entire modules
+iex> s Map
+@spec get(map(), key(), value()) :: value()
+@spec put(map(), key(), value()) :: map()
+# ...
+@spec get(map(), key()) :: value()
+@spec get_and_update!(map(), key(), (value() -> {get, value()})) :: {get, map()} | no_return() when get: term()
+```
+
+### `t`
+`t` Tells us about Types available in a given module:
+
+```elixir
+iex> t Map
+@type key() :: any()
+@type value() :: any()
+```
+
+And now we know that Map defines key and value types!
```

Comment:
Hmm, as someone who hasn't come accross `t` before, I still have no idea what it does. Would you mind explaining this further? 😄